api/app/signals/apps/api/ml_tool/utils.py

```diff
@@ -0,0 +1,35 @@
+from urllib.parse import urlparse
+
+from django.urls import resolve
+from rest_framework.reverse import reverse
+
+from signals.apps.signals.models import Category
+
+
+def get_category_from_prediction(category_url):
+    resolved = resolve(urlparse(category_url).path)
+    if not hasattr(resolved, 'kwargs'):
+        return category_url, False
+
+    slug = resolved.kwargs['sub_slug'] if 'sub_slug' in resolved.kwargs else resolved.kwargs['slug']
+
+    category = Category.objects.get(slug=slug)
```

Comment:
how do we deal with the `overig` problem?

Reply:
I think I removed this by accident. I added it again. Is solved in the V0 view.

Reply:
Do something with `parent__isnull=True` + `slug` or the like.

Reply:
👍